src/Forms/GridField/GridFieldGroupDeleteAction.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+/**
+ * Adds a delete action for the gridfield to remove a relationship from group.
+ * This is a special case where it captures whether the current user is the record being removed and
+ * prevents removal from happening.
+ *
+ * Class GroupGridFieldDeleteAction
+ * @package SilverStripe\Security
+ */
+class GridFieldGroupDeleteAction extends GridFieldDeleteAction
+{
+    /**
+     * @var integer
+     */
+    protected $userID;
+
+    public function __construct($userID = null)
+    {
+        $this->userID = $userID;
+
+        parent::__construct(true);
+    }
+
+    /**
+     *
+     * @param GridField $gridField
+     * @param DataObject $record
+     * @param string $columnName
+     * @return string the HTML for the column
+     */
+    public function getColumnContent($gridField, $record, $columnName)
+    {
+        $currentUser = Security::getCurrentUser();
+
+        if ($this->userID === $currentUser->ID || (
+            $record instanceof Member &&
+            $record->ID === $currentUser->ID
+        )) {
+            return null;
+        }
+
+        return parent::getColumnContent($gridField, $record, $columnName);
+    }
+
+    /**
+     * Handle the actions and apply any changes to the GridField
+     *
+     * @param GridField $gridField
+     * @param string $actionName
+     * @param mixed $arguments
+     * @param array $data - form data
+     * @throws ValidationException
+     */
+    public function handleAction(GridField $gridField, $actionName, $arguments, $data)
+    {
+        $currentUser = Security::getCurrentUser();
+
+        if ($actionName == 'unlinkrelation' && (
+            $this->userID === $currentUser->ID || (
+                $gridField->getList()->first() instanceof Member &&
+                $currentUser->ID === $arguments['RecordID']
+            )
+        )) {
+            throw new ValidationException(
+                _t(self::class . '.DeleteSelfFailure', 'Cannot remove yourself from groups')
```

Comment:
my understanding is the text collector cant handle `::class` and we have to use `__CLASS__` or strings

Reply:
yeah true, I forgot about `::class` not being supported

I'll include that in my rework